pyriemann/transferlearning.py

```diff
@@ -0,0 +1,175 @@
+from .utils.mean import mean_covariance
+from .utils.geodesic import geodesic
+
+
+class MDWM(MDM):
+    """Classification by Minimum Distance to Weighted Mean.
+
+    Classification by nearest centroid. For each of the given classes, a
+    centroid is estimated, according to the chosen metric, as a weighted mean
+    of covariance matrices from the source domain, combined with
+    the class centroid of the target domain [1]_ [2]_.
+    For classification, a given new point is attibuted to the class whose
+    centroid is the nearest according to the chosen metric.
+
+    Parameters
+    ----------
+    transfer_coef : float
+        Transfer coefficient in [0,1], controlling the trade-off between
+        source and target data. At 0, there is no transfer, only the data
+        acquired from the source are used. At 1, this is a calibration-free
+        system as no data are required from the source.
+    metric : string | dict (default: 'riemann')
+        The type of metric used for centroid and distance estimation.
+        see `mean_covariance` for the list of supported metric.
+        the metric could be a dict with two keys, `mean` and `distance` in
+        order to pass different metric for the centroid estimation and the
+        distance estimation. Typical usecase is to pass 'logeuclid' metric for
+        the mean in order to boost the computional speed and 'riemann' for the
+        distance in order to keep the good sensitivity for the classification.
+    n_jobs : int, (default: 1)
+        The number of jobs to use for the computation. This works by computing
+        each of the class centroid in parallel.
+        If -1 all CPUs are used. If 1 is given, no parallel computing code is
+        used at all, which is useful for debugging. For n_jobs below -1,
+        (n_cpus + 1 + n_jobs) are used. Thus for n_jobs = -2, all CPUs but one
+        are used.
+
+    Attributes
+    ----------
+    covmeans_ : list
+        the class centroids.
+    classes_ : list
+        list of classes.
+
+    See Also
+    --------
+    MDM
+
+    References
+    ----------
+    .. [1] E. Kalunga, S. Chevallier and Q. Barthelemy, "Transfer learning for
+        SSVEP-based BCI using Riemannian similarities between users", in 26th
+        European Signal Processing Conference (EUSIPCO), pp. 1685-1689. IEEE,
+        2018.
+
+    .. [2] S. Khazem, S. Chevallier, Q. Barthelemy, K. Haroun and C. Nous,
+        "Minimizing Subject-dependent Calibration for BCI with Riemannian
+        Transfer Learning", in 10th International IEEE/EMBS Conference on
+        Neural Engineering (NER), pp. 523-526. IEEE, 2021.
+    """
+
+    def __init__(self, transfer_coef, metric='riemann', n_jobs=1):
+        """Init."""
+        self.transfer_coef = transfer_coef
+        self.metric = metric
+        self.n_jobs = n_jobs
+        self.covmeans_ = None
+        self.classes_ = None
+        self._target_means = None
+        self._domain_means = None
```

Comment:
these attributes should only appear after the fit. This init should just set params as attributes